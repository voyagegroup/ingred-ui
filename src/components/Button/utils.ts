import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
