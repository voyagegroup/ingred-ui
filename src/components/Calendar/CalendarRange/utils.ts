import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
