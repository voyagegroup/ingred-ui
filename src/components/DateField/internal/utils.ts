export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
