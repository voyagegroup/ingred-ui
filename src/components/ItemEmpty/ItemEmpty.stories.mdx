import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import ItemEmpty from "./ItemEmpty";

<Meta title="Components/Utils/ItemEmpty" component={ItemEmpty} />

# ItemEmpty

<ArgsTable of={ItemEmpty} />

## Samples

<Canvas>
  <Story
    name="Example"
    args={{
      title: "There is not items.",
      subtitle: "this is subtitle.",
    }}
  >
    {(args) => <ItemEmpty {...args} />}
  </Story>
</Canvas>
