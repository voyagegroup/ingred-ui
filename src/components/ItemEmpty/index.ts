export { default } from "./ItemEmpty";
export type { ItemEmptyProps } from "./ItemEmpty";
