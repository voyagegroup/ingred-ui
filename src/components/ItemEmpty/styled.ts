import styled from "styled-components";

export const EmptyContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: ${({ theme }) => theme.spacing * 5}px
    ${({ theme }) => theme.spacing * 4}px ${({ theme }) => theme.spacing * 7}px;
`;

export const EmptyImage = styled.img<{ width: number }>`
  width: ${({ width }) => width}px;
  height: auto;
`;
