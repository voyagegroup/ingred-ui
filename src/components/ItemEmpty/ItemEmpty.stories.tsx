import React from "react";
import { StoryObj } from "@storybook/react";
import { Title, ArgsTable, Stories } from "@storybook/addon-docs";
import ItemEmpty, { ItemEmptyProps } from "./ItemEmpty";

export default {
  title: "Components/Utils/ItemEmpty",
  component: ItemEmpty,
  args: {
    title: "There is not items.",
    subtitle: "this is subtitle.",
  },
  parameters: {
    docs: {
      source: { language: "tsx" },
      page: () => (
        <>
          <Title />
          <ArgsTable of={ItemEmpty} />
          <Stories includePrimary title="Stories" />
        </>
      ),
    },
  },
};

export const Example: StoryObj<ItemEmptyProps> = {};
