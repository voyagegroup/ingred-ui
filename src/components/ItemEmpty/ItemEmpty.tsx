import * as React from "react";
import * as Styled from "./styled";
import Spacer from "../Spacer";
import Typography from "../Typography";
import defaultEmptyImage from "../../../assets/emptyImage.png";
import { useLocaleProps } from "../../hooks/useLocaleProps";

export type ItemEmptyProps = {
  title?: string;
  subtitle?: string;
  emptyImage?: string;
  imageWidth?: number;
};

const ItemEmpty = React.forwardRef<HTMLDivElement, ItemEmptyProps>(
  function ItemEmpty(inProps, ref) {
    const props = useLocaleProps({ props: inProps, name: "ItemEmpty" });
    const {
      title = "Not found.",
      subtitle,
      emptyImage,
      imageWidth = 80,
    } = props;

    return (
      <Styled.EmptyContainer ref={ref}>
        <Styled.EmptyImage
          src={emptyImage ? emptyImage : defaultEmptyImage}
          width={imageWidth}
        />
        <Spacer pt={2} pb={1}>
          <Typography weight="bold" size="md" align="center">
            {title}
          </Typography>
        </Spacer>
        {subtitle && <Typography size="sm">{subtitle}</Typography>}
      </Styled.EmptyContainer>
    );
  },
);

export default ItemEmpty;
