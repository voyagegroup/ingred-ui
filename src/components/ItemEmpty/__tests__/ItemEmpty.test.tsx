import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import ItemEmpty from "../ItemEmpty";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("ItemEmpty component testing", () => {
  afterEach(cleanup);

  test("ItemEmpty", () => {
    const { asFragment } = renderWithThemeProvider(
      <ItemEmpty title="title" subtitle="subtitle" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
