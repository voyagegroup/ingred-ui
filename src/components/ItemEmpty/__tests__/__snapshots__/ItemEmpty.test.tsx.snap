// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemEmpty component testing ItemEmpty 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL eAaefh"
  >
    <img
      class="sc-gsDKAQ dgYFcX"
      src=""
      width="80"
    />
    <div
      class="sc-dkPtRN ermvPK"
    >
      <p
        class="sc-hKwDye lbTzQi"
        color="#041C33"
        font-size="14px"
      >
        title
      </p>
    </div>
    <p
      class="sc-hKwDye dATcrn"
      color="#041C33"
      font-size="13px"
    >
      subtitle
    </p>
  </div>
</DocumentFragment>
`;
